content/guides/component/src/system_test.clj

```diff
@@ -0,0 +1,50 @@
+                                                                 ;; tag::ns[]
+(ns system-test
+  (:require [io.pedestal.http :as http]
+            [io.pedestal.http.route :as route]
+            [io.pedestal.test :refer [response-for]]
+            [com.stuartsierra.component :as component]
+            [clojure.test :refer :all]
+            [routes]
+            [system]
+            [pedestal]))
+                                                                 ;; end::ns[]
+
+                                                                 ;; tag::url-for[]
+(def url-for (route/url-for-routes
+              (route/expand-routes routes/routes)))
+                                                                 ;; end::url-for[]
+
+                                                                 ;; tag::service-fn[]
+(defn service-fn
+  [system]
+  (get-in system [:pedestal :service ::http/service-fn]))
+                                                                 ;; end::service-fn[]
+
+                                                                 ;; tag::with-system[]
+(defmacro with-system
+  [[bound-var binding-expr] & body]
+  `(let [~bound-var (component/start ~binding-expr)]
+     (try
+       ~@body
+       (finally
+         (component/stop ~bound-var)))))
+                                                                 ;; end::with-system[]
+
+                                                                 ;; tag::test-system[]
+(defn test-system
+  []
+  (system/app :pedestal-start-fn identity                        ;; <1>
+              :pedestal-stop-fn identity))                       ;; <2>
+                                                                 ;; end::test-system[]
+
+                                                                 ;; tag::test[]
+(deftest greeting-test
+  (with-system [sut (test-system)]                               ;; <1>
+    (let [service               (service-fn sut)                 ;; <2>
+          {:keys [status body]} (response-for service
```

Comment:
This is how we'd do it for real. For a guide, though, let's avoid destructuring in lets... I've noticed that it trips up beginners. It's OK to just do (is (= 200 (:status response))) in the body.